### PR TITLE
providers/implementations/exchange/kdf_exch.c: fix unavailable SIZE_MAX

### PR DESCRIPTION
**Pull request by levitte**
SIZE_MAX is used in a recent fix of this file, but without including
internal/numbers.h, so that macro ends up not existing on some platforms,
resulting in build failures.

### CODE REVIEW

File: providers/implementations/exchange/kdf_exch.c
Comment (t8m) [inline providers/implementations/exchange/kdf_exch.c:17]: you have some weird accented m here

**Review by t8m — CHANGES_REQUESTED** [review]

File: providers/implementations/exchange/kdf_exch.c
Comment (levitte) [reply providers/implementations/exchange/kdf_exch.c]: Fixed

**Review by t8m — APPROVED** [review]
Approved, assuming CI passes. I agree with urgent.

**Comment by levitte** [comment]
@openssl/committers, urgent second review needed

**Review by beldmit — APPROVED** [review]
LGTM, agree with urgent

**Review by paulidale — APPROVED** [review]

**Comment by levitte** [comment]
@mattcaswell, note that this PR is made against 3.0, so merge accordingly

**Comment by mattcaswell** [comment]
Pushed to master and 3.0.

**Pull request closed without merging**
